OneCommunityGlobal/HighestGoodNetworkApp
# Performance Optimization

# Description
Code changes to improve the front end application performance.
Made analysis on the code flow of Timelog, Summary Bar and Leaderboard component to identify candidates causing delayed loading of Dashboard/ Timelog pages.

## Mainly changes explained:
1. Remove console.log statements wherever possible while working on Dashboard and Timelog performance.
2. Import only required functions from "lodash" instead of entire library
3. Add overflow scroll to badges container as it overflows outside of the container.
4. Remove unused imports and variables
5. Lazy loaded badge images using HTML5 feature.
6. SummaryBar: useEffect has been updated to FETCH data from backend only if the requested page's "userId" is different from the authenticated user profile which is stored in redux global state. Else the already stored data is displayed.
7. SummaryBar: Moved statements from "return" to "useEffect()" wherever possible

## How to test:
check into current branch
do `npm install` and `...` to run this PR locally
Log in with differet role users
Check for general functioning of Dashboard, Timelog and Leaderboard.

## Note:
All the changes are a step closer to cleaner code and best practices. With multiple tests no considerable amount of performance improvement has been achieved. During this process, I have 3 more open points which when addressed will contribute for considerable performance. Will document the open points in the required docs.
